### PR TITLE
Adding a try catch handler to download data from alternative mirrors.

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
ref #1694

### CODE REVIEW

**Comment by sudk1896** [comment]
@martiansideofthemoon: Remove the astropy helpers commit, you would have trouble merging this later on.

**Comment by martiansideofthemoon** [comment]
@sudk1896 which commit specifically?

**Comment by sudk1896** [comment]
@martiansideofthemoon: Inside your commits, there is one commit called astropy helpers, never include it in your commits. You can do this by -
```
git reset --hard <tag/branch/commit_id>
git add .. #Add the files you want to commit, do **not include** astropy helpers.
git commit -m "your_message"
git push --force origin <your_branch_name>
```
I personally reset by commit_id. You can find that by doing ```git log``` on your current branch.

File: astropy_helpers [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 3e044eb26395c8569e3e15bf821c6ae5eeea324a
+Subproject commit fedd7c52ba80e2246f83b8d1752baa3a153a33e2
```
Comment (sudk1896) [inline astropy_helpers]: Remove this commit, entirely. It would mess with Travis.

**Comment by Cadair** [comment]
@sudk1896 That way of fixing it rewrites history, a better way to do it would be:

```
git checkout upstream/master astropy_helpers
git add astropy_helpers
git commit --amend
```
which should fix it.

**Comment by martiansideofthemoon** [comment]
@sudk1896 @Cadair done :)

**Comment by sudk1896** [comment]
@Cadair: Didn't know your method. @dpshelio had suggested me mine when I had done the same mistake in my PRs. Yours looks shorter too.

**Comment by martiansideofthemoon** [comment]
@Cadair @sudk1896 I can't seem to figure out why https://travis-ci.org/sunpy/sunpy/jobs/148414888 is failing :(

**Comment by dpshelio** [comment]
@sudk1896 I think it depends on when in history the commit your change is. Rewriting history is normally not good if other people are working with the particular branch you are changing. If not you can do so. I've just created issue #1858 to avoid this in the future

**Comment by dpshelio** [comment]
thanks @martiansideofthemoon and @abit2  - this looks cool! is there anyway we can add a test?

File: sunpy/data/_sample.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
 from sunpy.util.net import url_exists
 from sunpy import config
 
+import socket
```
Comment (dpshelio) [inline sunpy/data/_sample.py]: Since `socket` is a python library it should be placed in the first block, where `os`, `shutil`, etc...

**Comment by martiansideofthemoon** [comment]
@dpshelio better now?

**Comment by Cadair** [comment]
:+1:

**Comment by Cadair** [comment]
@dpshelio you good for merge on this?

**Comment by dpshelio** [comment]
:+1:

**Comment by martiansideofthemoon** [comment]
Thanks @Cadair ! What can I work on next?

**Pull request merged**